src/universal-dapp.js

```diff
@@ -650,6 +666,7 @@ UniversalDApp.prototype.runTx = function (data, args, cb) {
 
   var tx;
   if (!self.vm) {
+    self.transactionDebugger.switchProvider('EXTERNAL');
```

Comment:
Why do you switch the provider here and not at the point where the radiobutton is changed?